lib/features/manage_account/data/datasource_impl/notification_datasource_impl.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:permission_handler/permission_handler.dart';
+import 'package:tmail_ui_user/features/manage_account/data/datasource/notification_datasource.dart';
+import 'package:tmail_ui_user/features/manage_account/data/local/notification_setting_cache_manager.dart';
+import 'package:tmail_ui_user/features/manage_account/domain/exceptions/notification_exceptions.dart';
+import 'package:tmail_ui_user/main/exceptions/exception_thrower.dart';
+import 'package:tmail_ui_user/main/permissions/permission_service.dart';
+
+class NotificationDataSourceImpl implements NotificationDataSource {
+  final NotificationSettingCacheManager _notificationSettingCacheManager;
+  final PermissionService _permissionService;
+  final ExceptionThrower _exceptionThrower;
+
+  NotificationDataSourceImpl(
+    this._notificationSettingCacheManager,
+    this._permissionService,
+    this._exceptionThrower);
+
+  @override
+  Future<bool> getAppNotificationSetting() {
+    return Future.sync(() async {
+      final cache = _notificationSettingCacheManager.getAppNotificationSettingCache();
+      if (cache != null) return cache;
```

Comment:
What happens if user turn it on and store it in cache, then disable in system settings?

Reply:
For notification, system setting & cache setting will always be synced as soon as https://github.com/linagora/tmail-flutter/issues/2881 is implemented

Reply:
Cache temporarily turned off

Reply:
No notification can be shown if the system setting for notification is disabled